File: app/src/main/java/de/karzek/diettracker/presentation/main/diary/meal/adapter/favoriteRecipeList/viewHolder/FavoriteRecipeViewHolder.java
package de.karzek.diettracker.presentation.main.diary.meal.adapter.favoriteRecipeList.viewHolder;

import android.content.Context;
import android.graphics.BitmapFactory;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import com.daimajia.swipe.SwipeLayout;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.hdodenhof.circleimageview.CircleImageView;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.model.RecipeDisplayModel;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class FavoriteRecipeViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.favorite_recipe_image)
    CircleImageView imageView;
    @BindView(R.id.favorite_recipe_title)
    TextView title;

    private Context context;
    private final OnFavoriteRecipeItemClickedListener onItemClickedListener;

    public FavoriteRecipeViewHolder(ViewGroup viewGroup,
                                    OnFavoriteRecipeItemClickedListener onItemClickedListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_diary_favorite_recipe, viewGroup, false));
        ButterKnife.bind(this, itemView);

        context = viewGroup.getContext();
        this.onItemClickedListener = onItemClickedListener;
    }

    public void bind(RecipeDisplayModel entry) {
        if(entry.getPhoto() != null)
            imageView.setImageBitmap(BitmapFactory.decodeByteArray(entry.getPhoto(), 0, entry.getPhoto().length));
        else
            imageView.setImageDrawable(context.getDrawable(R.drawable.ic_chef_accent_dark));
        title.setText(entry.getTitle());

        itemView.setTag(entry.getId());
    }

    @OnClick(R.id.favorite_recipe_image) public void onItemClicked() {
        onItemClickedListener.onFavoriteRecipeItemClicked((int) itemView.getTag());
    }

    public interface OnFavoriteRecipeItemClickedListener {
        void onFavoriteRecipeItemClicked(int id);
    }
}

File: app/src/main/java/de/karzek/diettracker/presentation/main/diary/meal/adapter/favoriteRecipeList/viewHolder/FavoriteRecipeViewHolder.kt
package de.karzek.diettracker.presentation.main.diary.meal.adapter.favoriteRecipeList.viewHolder

import android.graphics.BitmapFactory
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.R
import de.karzek.diettracker.presentation.model.RecipeDisplayModel
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_diary_favorite_recipe.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class FavoriteRecipeViewHolder(override val containerView: View,
                               private val onItemClickedListener: OnFavoriteRecipeItemClickedListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(entry: RecipeDisplayModel) {
        if (entry.photo != null)
            favorite_recipe_image.setImageBitmap(BitmapFactory.decodeByteArray(entry.photo, 0, entry.photo.size))
        else
            favorite_recipe_image.setImageDrawable(containerView.context.getDrawable(R.drawable.ic_chef_accent_dark))
        favorite_recipe_title.text = entry.title

        itemView.tag = entry.id
        favorite_recipe_image.setOnClickListener { onItemClickedListener.onFavoriteRecipeItemClicked(itemView.tag as Int) }
    }

    interface OnFavoriteRecipeItemClickedListener {
        fun onFavoriteRecipeItemClicked(id: Int)
    }
}
